library/network_connections.py

```diff
@@ -805,6 +807,11 @@ def connection_create(self, connections, idx, connection_current=None):
             )
             s_wired = self.connection_ensure_setting(con, NM.SettingWired)
             s_wired.set_property(NM.SETTING_WIRED_MAC_ADDRESS, connection["mac"])
+            s_con.set_property(
+                NM.SETTING_CONNECTION_TYPE, NM.SETTING_MATCH_SETTING_NAME
+            )
+            s_match = self.connection_ensure_setting(con, NM.SettingMatch)
+            s_match.set_property(NM.SETTING_MATCH_PATH, connection["pci"])
```

Comment:
NetworkManager calls this property "path". It does so for a reason, namely, because it corresponds to udev's `ID_PATH` setting. In particular, it can also not be for the "pci" subsystem, but is more general. Likewise, systemd calls this "Path=" (see `man systemd.link`).

Either:

- the role also uses udev's `ID_PATH` parameter, so that it directly maps to what NetworkManager. In that case, the parameter in the role should not be named "pci", but something that follows udev and NetworkManager.

- be only the PCI address. In that case, the user would for example specify "0000:0a:00.0", and the role would then set "match.path=pci-0000:0a:00.0". But that seems not preferable, because it's more limited than the underlying setting.